packages/tasks/src/model-libraries-downloads.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * This file contains the (simplified) types used
+ * to represent queries that are made to Elastic
+ * in order to count number of model downloads
+ *
+ * Read this doc about download stats on the Hub:
+ *
+ * https://huggingface.co/docs/hub/models-download-stats
+ *
+ * see also:
+ * https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html
+ */
+
+export type ElasticBoolQueryFilter =
+	// match a single filename
+	| { term?: { path: string } }
+	// match multiple possible filenames
+	| { terms?: { path: string[] } }
+	// match a wildcard
+	| { wildcard?: { path: string } };
```

Comment:
(nit) FWIW, wildcards are not optimized in ES. Since we only use them to filter by suffix (`*.onnx`, `*.sentis`, etc.), we could use a [reverse-token-filter](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-reverse-tokenfilter.html) + a  [prefix filter](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-prefix-query.html).

Reply:
hmm maybe cc @SBrandeis or @co42 for information

Reply:
We would need to reindex everything to add a reverse token filter. If the wildcard is fast enough let's use that